src/types.ts

```diff
@@ -4324,6 +4324,14 @@ export interface StayComp extends Comp {
 	 * If the obj should not be destroyed on scene switch.
 	 */
 	stay: boolean,
+	/**
+	 * Scenes that either destroy the object or stay (depending on sceneStay value)
+	 */
+	scenes: string[],
+	/**
+	 * Whether or not the object stays on a scene or gets destroyed on a scene
+	 */
+	sceneStay: boolean,
 }
```

Comment:
Maybe `sceneStay` isn't needed. Like if you add the scenes props, it's because you want only stay in that scenes, and if not, simple stay in all scenes :)

Reply:
I thought it could be useful in the case that you wanted the **_ONLY_** delete the object if you are in those scenes instead of the other way around.

Reply:
some strange, I think it's more intuitive define the scenes for stay and not the scenes for not stay. What do you think @slmjkdbtl?

Reply:
The default is for the scenes to stay, you have an option to make it not stay

Reply:
thinking, maybe `stay(["level1", "level2"])` and `destroy(["level1", "level2"])` or `leave(["level1", "level2"])`

Perhaps that would be more intuitive, `not stay` is like contradicting the name of the component itself

Reply:
I think passing in a `false` there is more confusing to users than writing out all the scenes to stay for in the first array

Reply:
Yes, fs `stay(scenes: string[])` is the best option to avoid confusions

Reply:
So remove the `sceneStay` property, and add a `stayUntil(["level1"])` or similarly named component to have a similar effect while being more intuitive?

Reply:
nah, only `stay()` it's good for now :)

Reply:
Yeah I think one `stay(scenes: string[])` is great, thank you!!